controller/api/models.py

```diff
@@ -343,7 +343,7 @@ def deploy(self, user, release, initial=False):
             self._destroy_containers(existing)
 
         # perform default scaling if necessary
-        if initial:
+        if self.structure == {} and release.build is not None:
```

Comment:
Looks like `and release.build is not None` is additional logic that we didn't have before - is this intentional?